src/main/java/app/dbService/dao/BookDao.java

```diff
@@ -0,0 +1,61 @@
+package app.dbService.dao;
+
+import app.dbService.entity.Book;
+import app.dbService.executor.Executor;
+import app.dbService.executor.ResultHandler;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BookDao {
+    private final Executor executor;
+
+    public BookDao(Connection connection) {
+        this.executor = new Executor(connection);
+    }
+
+    public void insertBook(Book book) throws SQLException {
+        executor.execUpdate("INSERT INTO book (cover, title, authors, genres, publisher, publish_date, isbn, pages, description, total_amount, status) VALUES ('" + book.getCover64() + "' , '" + book.getTitle() + "', '" +
+                book.getAuthor() + "', '" + book.getGenre() + "', '" + book.getPublisher() + "', '" + book.getDate() + "', " +
+                book.getIsbn() + ", " + book.getPages() + ", '" + book.getDescription() + "', " + book.getAmount() + ", '" + book.getStatus() + "')");
+    }
+
+    public List<Book> listAllBooks() throws SQLException {
+        final List<Book> listBook = new ArrayList<>();
+
+        String sql = "SELECT * FROM book";
+        ResultHandler resultHandler = resultSet -> {
+            resultSet.next();
+            Book book = new Book(resultSet.getInt(1), resultSet.getString(3), resultSet.getString(4), resultSet.getString(7), resultSet.getInt(11));
+            listBook.add(book);
+            return listBook;
+        };
+
+        return (List<Book>) executor.execQuery(sql, resultHandler);
```

Comment:
you can get rid of cast with proper ResultHandler instantiation, as you did bellow